File: src/main/java/com/cocktailsguru/app/cocktail/dto/list/CocktailListResponseDto.java
package com.cocktailsguru.app.cocktail.dto.list;

import com.cocktailsguru.app.common.dto.PagingDto;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
public class CocktailListResponseDto {
    private List<CocktailListItemDto> cocktailList;
    private PagingDto pagingInfo;
}

File: src/main/java/com/cocktailsguru/app/cocktail/dto/list/CocktailListResponseDto.kt
package com.cocktailsguru.app.cocktail.dto.list

import com.cocktailsguru.app.cocktail.domain.CocktailList
import com.cocktailsguru.app.common.dto.ListResponseDto
import com.cocktailsguru.app.common.dto.PagingDto

data class CocktailListResponseDto(
        override val list: List<CocktailListItemDto>,
        override val pagingInfo: PagingDto
) : ListResponseDto<CocktailListItemDto> {
    constructor(cocktailList: CocktailList) : this(
            cocktailList.objectList.map { CocktailListItemDto(it) },
            PagingDto(cocktailList.pagingInfo)
    )
}